src/com/microsoftopentechnologies/aad/adal4j/AuthenticationContext.java

```diff
@@ -226,8 +225,9 @@ public AuthenticationResult acquireTokenByRefreshToken(
         int statusCode = connection.getResponseCode();
         if (statusCode != HttpURLConnection.HTTP_OK) {
             // TODO: Is IOException the right exception type to raise?
+            String err = CharStreams.toString(new InputStreamReader(connection.getErrorStream()));
```

Comment:
Shouldn't the stream returned by `connection.getErrorStream` and the new `InputStreamReader` object be closed? Maybe:

```
InputStream errorStream = null;
IntputStreamReader errorReader = null;
String err = "";
try {
  errorStream = connection.getErrorStream();
  errorReader = new InputStreamReader(errorStream);
  err = CharStreams.toString(errorReader);
}
finally {
  if(errorStream != null) {
    errorStream.close();
  }
  if(errorReader != null) {
    errorReader.close();
  }
}
```